gestalt-asset-core/src/main/java/org/terasology/gestalt/assets/AssetType.java

```diff
@@ -581,7 +656,7 @@ public boolean isLoaded(ResourceUrn urn) {
      * @return A list of all the loaded assets.
      */
     public Set<T> getLoadedAssets() {
-        return ImmutableSet.copyOf(loadedAssets.values());
+        return ImmutableSet.copyOf(loadedAssets.values().stream().map(Reference::get).filter(Objects::nonNull).collect(Collectors.toSet()));
```

Comment:
There's no need to make an ImmutableSet copy and them stream the values into a new set could just stream into the set.